### PR TITLE
Add more documentation to hist and barh

### PR DESCRIPTION
**Pull request by SamLau95**
Also makes `Table.hist` and `Table.barh` throw a more helpful error message when the
Table contains invalid values (`np.flexible`, which includes strings).
Before, the method raised an error about not being able to operate on
"flexible types", which is not extremely helpful without knowledge of
numpy types.

Starts to address #50.

Travis tests are failing because of a long-standing issue with the date formatter.

### CODE REVIEW

**Comment by SamLau95** [comment]
@papajohn @alvinwan Either one of you wanna give this a review?

File: datascience/tables.py [hunk 1]
```diff
@@ -582,10 +582,82 @@ def annotate(axis, ticks):
...
+        A total of n - 1 charts are created where n is the number of columns
+        in the table.
+
+        Note that the order of the categories in the table is not guaranteed to
+        be preserved in the bar graph. Ex. `barh` on a table with "a", "b", "c"
+        as the rows in the `column_for_categories` may not output a bar graph
+        with the labels in that order.
+
+        Requires every column except for `column_for_categories` to be
+        numberical. If the columns contain other types, a `ValueError` is
```
Comment (papajohn) [inline datascience/tables.py]: numerical

File: datascience/tables.py [hunk 2]
```diff
@@ -582,10 +582,82 @@ def annotate(axis, ticks):
...
+
+        Each chart is categorized using the values in `column_for_categories`
+        and one chart is produced for every other column in the table.
+        A total of n - 1 charts are created where n is the number of columns
+        in the table.
+
+        Note that the order of the categories in the table is not guaranteed to
+        be preserved in the bar graph. Ex. `barh` on a table with "a", "b", "c"
+        as the rows in the `column_for_categories` may not output a bar graph
+        with the labels in that order.
```
Comment (papajohn) [inline datascience/tables.py]: Maybe we should just fix this behavior... filing an issue.

**Comment by papajohn** [comment]
Yes, that's much improved!

**Pull request merged**